astropy/io/ascii/cds.py

```diff
@@ -21,6 +21,27 @@
 __doctest_skip__ = ["*"]
 
 
+def _is_section_delimiter(line):
+    """Check if line is a section delimiter
+
+    CDS/MRT tables use dashes or equal signs ("------" or "======") to
+    separate sections. This function checks if a line contains only either
+    of these characters.
+
+    Parameters
+    ----------
+    line : str
+        String containing an entire line from the table text file.
+
+    Returns
+    -------
+    bool
+        True if the line is a section delimiter, False otherwise.
+
+    """
+    return line.startswith(("------", "=======")) and len(set(line)) == 1
```

Comment:
Since the CDS/MRT specs are not super precise, it might be good to protect against trailing whitespace after a section header. So add this change and put in some trailing whitespace in the test file.
```suggestion
    return line.startswith(("------", "=======")) and len(set(line.strip())) == 1
```